// @smart-webcomponents-angular/window/source/modules/smart.window.js

/* Smart UI v15.0.2 (2023-01-24) 
Copyright (c) 2011-2023 jQWidgets. 
License: https://htmlelements.com/license/ */ //

import '../../common/runtime.js';
import "../smart.element.js";
import "../smart.button.js";
import "../smart.scrollbar.js";
import "../smart.listbox.js";
import "../smart.dropdownlist.js";
import "../smart.combobox.js";
import "../smart.textbox.js";
import "../smart.multilinetextbox.js";
import "../smart.progressbar.js";
import "../smart.window.js";

export class smartWindow extends Smart.Component {
	get name() {
		return 'smartWindow';
	}		
}

export class smartDialogWindow extends Smart.Component {
	get name() {
		return 'smartDialogWindow';
	}		
}

export class smartAlertWindow extends Smart.Component {
	get name() {
		return 'smartAlertWindow';
	}		
}

export class smartPromptWindow extends Smart.Component {
	get name() {
		return 'smartPromptWindow';
	}		
}

export class smartMultiLinePromptWindow extends Smart.Component {
	get name() {
		return 'smartMultiLinePromptWindow';
	}		
}

export class smartProgressWindow extends Smart.Component {
	get name() {
		return 'smartProgressWindow';
	}		
}

export class smartTabsWindow extends Smart.Component {
	get name() {
		return 'smartTabsWindow';
	}		
}

export class smartWaitWindow extends Smart.Component {
	get name() {
		return 'smartWaitWindow';
	}		
}
